codebase/app/rest_server/rest_server.go

```diff
@@ -86,10 +92,44 @@ func NewServer(service factory.ServiceFactory, opts ...OptionFunc) factory.AppSe
 
 func (s *restServer) Serve() {
 	var err error
-	if s.listener != nil {
-		err = s.httpEngine.Serve(s.listener)
+
```

Comment:
jadi ketika `Serve` cukup begini aja
```suggestion
func (s *restServer) Serve() {
	var err error

	if s.listener == nil {
		s.listener, err = net.Listen("tcp", s.httpEngine.Addr)
		if err != nil {
			log.Panicf("REST TCP Listener: Unexpected Error: %v", err)
		}
	}

	if s.opt.tlsConfig != nil {
		s.httpEngine.TLSConfig = s.opt.tlsConfig
		s.listener = tls.NewListener(s.listener, s.opt.tlsConfig)
	}

	err = s.httpEngine.Serve(s.listener)
	switch err.(type) {
	case *net.OpError:
		log.Panicf("REST Server: Unexpected Error: %v", err)
	}
}
```